test/test_nested_attributes_2d.jl

```diff
@@ -18,7 +18,7 @@ end
 
 function refine_fn(p4est, which_tree, quadrant)
   quadrant_obj = unsafe_load(quadrant)
-  if quadrant_obj.x == 0 && quadrant_obj.y == 0 && quadrant_obj.level < 4
+  if quadrant_obj.x == 0 && quadrant_obj.y == 0 && quadrant_obj.level < 2
```

Comment:
Why is this change necessary?

Reply:
Otherwise, we get  [this](https://github.com/trixi-framework/P4est.jl/actions/runs/6523830565/job/17714856539?pr=90#step:11:643) error, i.e. [this assertion in p4est](https://github.com/cburstedde/p4est/blob/master/src/p4est_iterate.c#L2347) fails. I am not an expert of the internals of p4est to fully explain *why* this fails, however balancing the forest before iterating over it, fixes the problem (see [ae6e0d3](https://github.com/trixi-framework/P4est.jl/pull/98/commits/ae6e0d322c21932f901733d2382f1b7dff1336aa)) and is probably the proper fix. Iterating over a forest with neighboring trees that differ in more than level does not work (or something along those lines).

Reply:
Thanks 👍